contracts/contracts/vault/OETHVaultCore.sol

```diff
@@ -125,24 +137,338 @@ contract OETHVaultCore is VaultCore {
+            // Need to remove WETH that is reserved for the withdrawal queue
+            if (balance + queue.claimed >= queue.queued) {
+                return balance + queue.claimed - queue.queued;
+            }
+        }
+    }
+
+    /**
+     * @notice Allocate unallocated funds on Vault to strategies.
+     **/
+    function allocate() external override whenNotCapitalPaused nonReentrant {
+        // Add any unallocated WETH to the withdrawal queue first
+        _addWithdrawalQueueLiquidity();
+
+        _allocate();
+    }
+
+    /// @dev Allocate WETH to the default WETH strategy if there is excess to the Vault buffer.
+    /// This is called from either `mint` or `allocate` and assumes `_addWithdrawalQueueLiquidity`
+    /// has been called before this function.
+    function _allocate() internal override {
+        // No need to do anything if no default strategy for WETH
+        address depositStrategyAddr = assetDefaultStrategies[weth];
+        if (depositStrategyAddr == address(0)) return;
+
+        uint256 wethAvailableInVault = _wethAvailable();
+        // No need to do anything if there isn't any WETH in the vault to allocate
+        if (wethAvailableInVault == 0) return;
+
+        // Calculate the target buffer for the vault using the total supply
+        uint256 totalSupply = oUSD.totalSupply();
+        uint256 targetBuffer = totalSupply.mulTruncate(vaultBuffer);
+
+        // If available WETH in the Vault is below or equal the target buffer then there's nothing to allocate
+        if (wethAvailableInVault <= targetBuffer) return;
+
+        // The amount of assets to allocate to the default strategy
+        uint256 allocateAmount = wethAvailableInVault - targetBuffer;
+
+        IStrategy strategy = IStrategy(depositStrategyAddr);
+        // Transfer WETH to the strategy and call the strategy's deposit function
+        IERC20(weth).safeTransfer(address(strategy), allocateAmount);
+        strategy.deposit(weth, allocateAmount);
+
+        emit AssetAllocated(weth, depositStrategyAddr, allocateAmount);
+    }
+
+    /// @dev The total value of all assets held by the vault and all its strategies
+    /// less any WETH that is reserved for the withdrawal queue.
+    /// For OETH, this is just WETH in the vault and strategies.
+    ///
+    // If there is not enough WETH in the vault and all strategies to cover all outstanding
+    // withdrawal requests then return a total value of 0.
+    function _totalValue() internal view override returns (uint256 value) {
+        value = _totalValueInVault() + _totalValueInStrategies();
+
+        // Need to remove WETH that is reserved for the withdrawal queue.
+        WithdrawalQueueMetadata memory queue = withdrawalQueueMetadata;
+        // reserved for the withdrawal queue = cumulative queued total - total claimed
+        uint256 reservedForQueue = queue.queued - queue.claimed;
+
+        if (reservedForQueue > 0) {
```

Comment:
If `reservedForQueue` is zero, then the rest of everything here just works fine, with no need to special case it? Can we lose this outer level if statement?

Secondly, I like having the exceptions checks return out early, and then have the success path flow through the function to the final return. This feels a little awkward here having the success return ahead of the special case 0 return.

Reply:
If `reservedForQueue` is zero then we want to return the value from `_totalValueInVault() + _totalValueInStrategies()`. If we do both your changes then the code is simpler and easier to reason.

```Solidity
function _totalValue() internal view override returns (uint256 value) {
    value = _totalValueInVault() + _totalValueInStrategies();

    // Need to remove WETH that is reserved for the withdrawal queue.
    WithdrawalQueueMetadata memory queue = withdrawalQueueMetadata;
    // reserved for the withdrawal queue = cumulative queued total - total claimed
    uint256 reservedForQueue = queue.queued - queue.claimed;

    if (value < reservedForQueue) {
        // This can happen if the vault becomes insolvent enough that the
        // total value in the vault and all strategies < outstanding withdrawals.
        // For example, there was a mass slashing event and most users request a withdrawal.
        return 0;
    }

    // Adjust the total value by the amount reserved for the withdrawal queue
    return value - reservedForQueue;
}
```